src/components/tasks/Monday.tsx

```diff
@@ -0,0 +1,327 @@
+import { TreeItem, TreeView } from "@mui/lab";
+import {
+  Autocomplete,
+  Box,
+  ClickAwayListener,
+  MenuItem,
+  Paper,
+  Stack,
+  StackProps,
+  TextField,
+  Typography,
+} from "@mui/material";
+import { useEffect, useState } from "react";
+import { Controller, useFormContext } from "react-hook-form";
+import { StyledTreeItem } from "src/components/formComponents/styledTreeItem";
+import { graphql } from "src/generated";
+import {
+  MondayFolderColorEnum,
+  MutationCreateMondayItemArgs,
+} from "src/generated/graphql";
+
+interface Group {
+  id: string;
+  name: string;
+  board?: Board;
+}
+
+interface PaginatedGroups {
+  elements?: Group[] | null | undefined;
+}
+
+interface Board {
+  id: string;
+  name: string;
+}
+
+interface PaginatedBoards {
+  elements?: (Board & { groups: PaginatedGroups })[] | null | undefined;
+}
+
+interface PaginatedFolders {
+  elements?:
+    | {
+        id: string;
+        name: string;
+        color?: MondayFolderColorEnum | null | undefined;
+        boards: PaginatedBoards;
+        folders?: PaginatedFolders;
+      }[]
+    | null
+    | undefined;
+}
+
+interface MondayProps extends StackProps {
+  userModuleId: string;
+  highlightedText: string;
+}
+
+const QUERY_MONDAY_WORKSPACES = graphql(/* GraphQL */ `
+  query MondayWorkspaces($userModuleId: ID!) {
+    mondayWorkspaces(userModuleId: $userModuleId) {
+      elements {
+        id
+        name
+      }
+    }
+  }
+`);
+
+const QUERY_MONDAY_RESOURCES = graphql(/* GraphQL */ `
+  query MondayResources($userModuleId: ID!, $workspaceId: ID!) {
+    mondayResources(userModuleId: $userModuleId, workspaceId: $workspaceId) {
+      boards {
+        elements {
+          id
+          name
+          groups {
+            elements {
+              id
+              name
+            }
+          }
+        }
+      }
+      folders {
+        elements {
+          id
+          name
+          color
+          boards {
+            elements {
+              id
+              name
+              groups {
+                elements {
+                  id
+                  name
+                }
+              }
+            }
+          }
+          folders {
+            elements {
+              id
+              name
+              color
+              boards {
+                elements {
+                  id
+                  name
+                  groups {
+                    elements {
+                      id
+                      name
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+`);
+
+const Monday = (props: MondayProps) => {
+  const { userModuleId, highlightedText } = props;
+  const { register, setValue, control } =
+    useFormContext<MutationCreateMondayItemArgs>();
+
+  const [selectedWorkspace, setSelectedWorkspace] = useState("");
+  const [selectedGroup, setSelectedGroup] = useState<Group>({
```

Comment:
Could this be `Group | null` instead of having an empty object?

Reply:
If object is null, the autocomplete doesn’t work when user select group.
Do you know another way to fix that?

Reply:
Ok, my bad, it doesn't occur actually.

The only problem is autocomplete can't have null.
And if I set undefined, we got a warning on the autocomplete (about it's not controlled component and suddenly it becomes controlled, something like that ^^).

So I set null and default value {id: null, name: null}